concepts/basics/about.md

```diff
@@ -0,0 +1,88 @@
+# About
+
+PowerShell is a dynamically implicit typed (also support strongly typed variables) scripting language from Microsoft, mostly focus on task automation and confifuration management.
+
+PowerShell supports imperative, functional, object-oriented programming, and even pipeline style.
+
+PowerShell was built on the .NET platform and first released in 2006.
+It has been included by default in all modern Windows OS, and can be installed on Mac and Linux.
+
+## Variables
+
+There are two common ways to define a variable in PowerShell.
+
+Varible name in PowerShell must start with `$` and using the assignment operator `=` to bind the variable name to a value.
+
+```powershell
+> $intVar = 1
+> $doubleVar = 2.0
+> $stringVar = "Hello"
+
+# Checking a variable type using the GetType() method
+> Write-Host $doubleVar.GetType()
+System.Double
+```
+
+You can also explicitly specify a variable's type if you want to enforce a specific type.
+PowerShell will try its best to convert the value from one type to another if possible.
+
+```powershell
+# String represents a floating number can be converted to integer
+> [int]$staticOne = "1.234"
+> $staticOne
+1
+
+# String of letters can not be converted to double type
+> [double]$staticTwo = "hello"
+MetadataError: Cannot convert value "hello" to type "System.Double"
+```
+
+### Constants
+
+Constants in PowerShell are read-only variables that can't be **removed** after defined.
+This lead us to the second way to define variables using the cmdlet `New-Variable`.
+
+```powershell
+> New-Variable -Name "MyConstant" -Value 100 -Option Constant
+> $MyConstant
+100
+```
+
+For a more detailed and advanced way to create a new variable with various options, please read [New-Variable][New-Variable].
+
+## Functions
+
+Functions in PowerShell are defined using the `function` keyword.
+Each function can have zero or more formal parameters in `()` parenthesis, followed by its own scope start  with `{` and end with `}`. 
+
+```powershell
+function Greeting {
+    return "Hello!"
+}
+```
+
+Functions in PowerShell can return values explicitly via the `return` keyword, after being called using the syntax `FuncName $Arg`.
+
+```powershell
+function AddTwoNumbers($NumOne, $NumTwo):
+    $result = $NumOne + $NumTwo
+    return $result
+
+> AddTwoNumbers 10 20
+30
+```
+
+## Comments
+
+Single comment in PowerShell start with a `#`, and end at line termination.
+Comment block in PowerShell start with `<#` and end with `#>`.
+
+```powershell
+function Main() {
+    #This is a comment
+}
+```
```

Comment:
Feels like there should also be a mulit-line example here, to demonstrate what that looks like.